frontend/src/plugins/impl/MultiselectPlugin.tsx

```diff
@@ -67,18 +86,25 @@ const Multiselect = (props: MultiselectProps): JSX.Element => {
         displayValue={(option) => option}
         placeholder="Select..."
         multiple={true}
-        filterFn={multiselectFilterFn}
         className={cn({
           "w-full": props.fullWidth,
         })}
         value={props.value}
         onValueChange={(newValues) => setValue(newValues || [])}
+        shouldFilter={false}
+        customFilterFn={filterOptions}
       >
-        {props.options.map((option) => (
-          <ComboboxItem key={option} value={option}>
-            {option}
-          </ComboboxItem>
-        ))}
+        {/* List virtualization */}
+        <Virtuoso
+          style={{ height: "200px" }}
+          totalCount={filteredOptions.length}
+          overscan={25}
+          itemContent={(i: number) => (
+            <ComboboxItem key={filteredOptions[i]} value={filteredOptions[i]}>
+              {filteredOptions[i]}
+            </ComboboxItem>
+          )}
+        />
```

Comment:
did you notice any downside if the size is 10 (and we don't need to virtualize?)
if there is no downside, we can always virtualize

but if it feels worse for small numbers, we do do something like:

```ts
const renderList = () {

  if (options.length > 100) {
    return <Virtuoso
  }
  return  props.options.map((option) => (
          <ComboboxItem key={option} value={option}>
            {option}
          </ComboboxItem>
        ))}
}
```

Reply:
Just checked with 50 items and it feels like the list is sort of jumping when I scroll. I think this is a good idea.

Reply:
cool - ill defer to you on the number. maybe 200?

Reply:
I think 200 is good. To some extent, there is some stuttering / flickering with 10,000 items if you do a big scroll and let the list slow to a stop. But I don't think it's too noticeable with controlled scrolls, and better than a laggy list / bloated DOM anyway.

Reply:
I'll commit my test notebook in a sec and let you be the judge